core/ViewDataTable/Config.php

```diff
@@ -515,6 +527,12 @@ public function __construct()
             Metrics::getDefaultProcessedMetrics()
         );
 
+        $periodValidator = new PeriodValidator();
+        $this->selectable_periods = $periodValidator->getPeriodsAllowedForUI();
+        $this->selectable_periods = array_diff($this->selectable_periods, array('range'));
```

Comment:
Should this filtering also be done before rendering? For example, if someone sets the periods to `['day', 'week', 'garbage', 'range']`, should we remove the values that aren't allowed?

Reply:
I would have said it's not needed maybe cause a plugin might support it but not actually implement it fully in Matomo but that shouldn't be an issue I suppose. I think the feature usually won't be used so wouldn't do too much about it.